src/coreclr/pal/src/exception/signal.cpp

```diff
@@ -344,6 +344,26 @@ bool IsRunningOnAlternateStack(void *context)
 #endif // HAVE_MACH_EXCEPTIONS
 }
 
+static bool IsSaSigInfo(struct sigaction* action)
+{
+    return (action->sa_flags & SA_SIGINFO) != 0;
+}
+
+static bool IsSigDfl(struct sigaction* action)
+{
+    // macOS can return sigaction with SIG_DFL and SA_SIGINFO.
+    // SA_SIGINFO means we should use sa_sigaction, but here we want to check sa_handler.
+    // So we ignore SA_SIGINFO when sa_sigaction and sa_handler are at the same address.
+    return (&action->sa_handler == (void*)&action->sa_sigaction || !IsSaSigInfo(action)) &&
```

Comment:
A nit - can you please add `( )` to explicitly express the precedence in all the composite conditions? It is a preferred style in this codebase. Like this:
```
return ((&action->sa_handler == (void*)&action->sa_sigaction) || !IsSaSigInfo(action)) &&
(action->sa_handler == SIG_DFL));
```